tests/audit/report_test.py

```diff
@@ -0,0 +1,128 @@
+    filename = baseline_file()
+    output = generate_report(filename, class_to_print)
+    real, false = count_results(output)
+    assert real == expected_real
+    assert false == expected_false
+    delete_all_temporal_files()
+
+
+def count_results(data):
+    real_secrets = 0
+    false_secrets = 0
+    for secret in data:
+        if SecretClassToPrint.from_class(VerifiedResult[secret['category']]) == SecretClassToPrint.REAL_SECRET:  # noqa: E501
+            real_secrets += 1
+        else:
+            false_secrets += 1
+    return real_secrets, false_secrets
+
+
+def baseline_file():
+    # Create our own SecretsCollection manually, so that we have fine-tuned control.
+    url_format = 'http://username:{}@www.example.com/auth'
+    first_secret = 'value1'
+    second_secret = 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lIiwiaWF0IjoxNTE2MjM5MDIyfQ'  # noqa: E501
+    random_secret = ''.join(random.choice(string.ascii_letters) for _ in range(8))
+    file_content = 'url = ' + url_format.format(first_secret) \
+        + '\nexample = ' + url_format.format(random_secret) \
+        + '\nlink = ' + url_format.format(first_secret)
+    first_file = create_file_with_content(file_content)
+    file_content = 'url = ' + url_format.format(second_secret) \
+        + '\nexample = ' + url_format.format(random_secret)
+    second_file = create_file_with_content(file_content)
+    secrets = SecretsCollection()
+    secrets[first_file] = {
+        original_potential_secret_factory(
+            type=BasicAuthDetector.secret_type,
+            secret=first_secret,
+            is_secret=True,
+            line_number=1,
+            filename=first_file,
+        ),
+        original_potential_secret_factory(
+            type=BasicAuthDetector.secret_type,
+            secret=random_secret,
+            is_secret=False,
+            line_number=2,
+            filename=first_file,
+        ),
+        original_potential_secret_factory(
+            type=BasicAuthDetector.secret_type,
+            secret=first_secret,
+            is_secret=True,
+            line_number=3,
+            filename=first_file,
+        ),
+    }
+    secrets[second_file] = {
+        original_potential_secret_factory(
+            type=JwtTokenDetector.secret_type,
+            secret=second_secret,
+            is_secret=True,
+            line_number=1,
+            filename=second_file,
+        ),
+        original_potential_secret_factory(
+            type=BasicAuthDetector.secret_type,
+            secret=second_secret,
+            is_secret=False,
+            line_number=1,
+            filename=second_file,
+        ),
+        original_potential_secret_factory(
+            type=BasicAuthDetector.secret_type,
+            secret=random_secret,
+            is_secret=False,
+            line_number=2,
+            filename=second_file,
+        ),
+    }
```

Comment:
Interesting. Why not just do:

```python
secrets.scan_file(first_file)
secrets.scan_file(second_file)
```

if you need to label them later, you can do:

```python
labels = [True, False, None, ...]
for label, item in zip(labels, secrets):
    _, secret = item
    secret.is_secret = label
```

Probably more resilient (and more realistic test) than hard-coding, especially since you already went through the trouble of crafting secrets that would be caught by the plugins.